### PR TITLE
client/request: wrap some errors

### PR DESCRIPTION
**Pull request by thaJeztah**
Just a small nit; try to preserve original errors where possible 👍

### CODE REVIEW

**Comment by olljanat** [comment]
New error message on Z:
```
14:35:31 Wiping out workspace first.
14:36:37 java.nio.file.AccessDeniedException: /home/jenkins/workspace/Docker-PRs-s390x/bundles/test-integration/d6e165cb80b5c/docker.log
14:36:37 	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:84)
14:36:37 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
14:36:37 	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
14:36:37 	at sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:244)
14:36:37 	at sun.nio.fs.AbstractFileSystemProvider.deleteIfExists(AbstractFileSystemProvider.java:108)
14:36:37 	at java.nio.file.Files.deleteIfExists(Files.java:1165)
14:36:37 	at hudson.Util.tryOnceDeleteFile(Util.java:290)
14:36:37 	at hudson.Util.deleteFile(Util.java:245)
14:36:37 Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from ip-172-31-0-216.us-west-2.compute.internal/172.31.0.216:18521
14:36:37 		at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1693)
14:36:37 		at hudson.remoting.UserResponse.retrieve(UserRequest.java:310)
14:36:37 		at hudson.remoting.Channel.call(Channel.java:908)
14:36:37 		at hudson.FilePath.act(FilePath.java:986)
14:36:37 		at hudson.FilePath.act(FilePath.java:975)
14:36:37 		at hudson.FilePath.deleteContents(FilePath.java:1190)
14:36:37 		at hudson.plugins.git.extensions.impl.WipeWorkspace.beforeCheckout(WipeWorkspace.java:30)
14:36:37 		at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1157)
14:36:37 		at hudson.scm.SCM.checkout(SCM.java:495)
14:36:37 		at hudson.model.AbstractProject.checkout(AbstractProject.java:1202)
14:36:37 		at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:574)
14:36:37 		at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
14:36:37 		at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:499)
14:36:37 		at hudson.model.Run.execute(Run.java:1724)
14:36:37 		at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
14:36:37 		at hudson.model.ResourceController.execute(ResourceController.java:97)
14:36:37 		at hudson.model.Executor.run(Executor.java:429)
14:36:37 Caused: java.io.IOException: Unable to delete '/home/jenkins/workspace/Docker-PRs-s390x/bundles/test-integration/d6e165cb80b5c/docker.log'. Tried 3 times (of a maximum of 3) waiting 0.1 sec between attempts.
14:36:37 	at hudson.Util.deleteFile(Util.java:250)
14:36:37 	at hudson.FilePath.deleteRecursive(FilePath.java:1210)
14:36:37 	at hudson.FilePath.deleteContentsRecursive(FilePath.java:1219)
14:36:37 	at hudson.FilePath.deleteRecursive(FilePath.java:1201)
14:36:37 	at hudson.FilePath.deleteContentsRecursive(FilePath.java:1219)
14:36:37 	at hudson.FilePath.deleteRecursive(FilePath.java:1201)
14:36:37 	at hudson.FilePath.deleteContentsRecursive(FilePath.java:1219)
14:36:37 	at hudson.FilePath.deleteRecursive(FilePath.java:1201)
14:36:37 	at hudson.FilePath.deleteContentsRecursive(FilePath.java:1219)
14:36:37 	at hudson.FilePath.access$1100(FilePath.java:196)
14:36:37 	at hudson.FilePath$15.invoke(FilePath.java:1193)
14:36:37 	at hudson.FilePath$15.invoke(FilePath.java:1190)
14:36:37 	at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2760)
14:36:37 	at hudson.remoting.UserRequest.perform(UserRequest.java:207)
14:36:37 	at hudson.remoting.UserRequest.perform(UserRequest.java:53)
14:36:37 	at hudson.remoting.Request$2.run(Request.java:358)
14:36:37 	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
14:36:37 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
14:36:37 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
14:36:37 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
14:36:37 	at hudson.remoting.Engine$1$1.run(Engine.java:98)
14:36:37 	at java.lang.Thread.run(Thread.java:748)
```
but let's try once again...

**Comment by olljanat** [comment]
Derek add label: rebuild/z

File: client/request.go [hunk 1]
```diff
@@ -220,14 +220,14 @@ func (cli *Client) checkResponseErr(serverResp serverResponse) error {
...
 		}
-		errorMessage = errorResponse.Message
+		errorMessage = strings.TrimSpace(errorResponse.Message)
 	} else {
-		errorMessage = string(body)
+		errorMessage = strings.TrimSpace(string(body))
 	}
 
-	return fmt.Errorf("Error response from daemon: %s", strings.TrimSpace(errorMessage))
+	return errors.Wrap(errors.New(errorMessage), "Error response from daemon")
```
Comment (AkihiroSuda) [inline client/request.go]: `errors.Wrap(errors.New("error response from daemon", errorMessasge))`?

**Review by AkihiroSuda — APPROVED** [review]

**Review by vdemeester — APPROVED** [review]
LGTM :tiger: 
/cc @cpuguy83

**Pull request merged**